src/App.js

```diff
@@ -30,25 +29,76 @@ const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [currPlayer, setCurrentPlayer] = useState(PLAYER_1);
+  const [currWinner, setWinner] =useState('No one yet')
+
+  const updateSquareData = (updatedSquare) => {
+
+    // const flatarr = squares.flat();
+    const newSquares = squares.map(row =>{
+      for (let j =0; j<row.length; j++){
+        const square = row[j]
+        if (square.id === updatedSquare.id) {
+          square.value = currPlayer;
+        }  
+      }
+      return row;    
+    });
```

Comment:
This would be clearer as a nested map:

```suggestion
    const newSquares = squares.map(row =>{
      return row.map(square => {
        if (square.id === updatedSquare.id) {
          return {...square, value: currPlayer};
        }  else {
          return square;
        }
      });
    });
```